samples/webassembly/Triangle/Triangle.cpp

```diff
@@ -0,0 +1,254 @@
+    }
+  });
+
+  // @lint-ignore CLANGTIDY
+  glfwSetWindowSizeCallback(window, [](GLFWwindow* /*window*/, int width, int height) {
+    printf("Window resized! width=%d, height=%d\n", width, height);
+    width_ = width;
+    height_ = height;
+  });
+
+  glfwGetWindowSize(window, &width_, &height_);
+
+  if (outWindow) {
+    *outWindow = window;
+  }
+
+  return true;
+}
+
+static void initIGL() {
+  // create a device
+  {
+    auto ctx =
+        std::make_unique<igl::opengl::webgl::Context>(igl::opengl::RenderingAPI::GLES3, "#canvas");
+    device_ = std::make_unique<igl::opengl::webgl::Device>(std::move(ctx));
+
+    IGL_ASSERT(device_);
+  }
+
+  // Command queue: backed by different types of GPU HW queues
+  CommandQueueDesc desc{CommandQueueType::Graphics};
+  commandQueue_ = device_->createCommandQueue(desc, nullptr);
+
+  // Color attachment
+  renderPass_.colorAttachments[0] = igl::RenderPassDesc::ColorAttachmentDesc{};
+  renderPass_.colorAttachments[0].loadAction = LoadAction::Clear;
+  renderPass_.colorAttachments[0].storeAction = StoreAction::Store;
+  renderPass_.colorAttachments[0].clearColor = {1.0f, 1.0f, 1.0f, 1.0f};
+  renderPass_.depthAttachment.loadAction = LoadAction::DontCare;
+}
+
+static void createRenderPipeline() {
+  if (renderPipelineState_Triangle_) {
+    return;
+  }
+
+  IGL_ASSERT(framebuffer_);
+
+  RenderPipelineDesc desc;
+
+  desc.targetDesc.colorAttachments.resize(1);
+
+  // @fb-only
+  if (framebuffer_->getColorAttachment(0)) {
+    desc.targetDesc.colorAttachments[0].textureFormat =
+        framebuffer_->getColorAttachment(0)->getFormat();
+  }
+
+  if (framebuffer_->getDepthAttachment()) {
+    desc.targetDesc.depthAttachmentFormat = framebuffer_->getDepthAttachment()->getFormat();
+  }
+
+  desc.shaderStages = ShaderStagesCreator::fromModuleStringInput(
+      *device_, codeVS.c_str(), "main", "", codeFS, "main", "", nullptr);
+  renderPipelineState_Triangle_ = device_->createRenderPipeline(desc, nullptr);
+  IGL_ASSERT(renderPipelineState_Triangle_);
+}
+
+static std::shared_ptr<ITexture> getNativeDrawable() {
+  Result ret;
+  std::shared_ptr<ITexture> drawable;
+  const auto& platformDevice = device_->getPlatformDevice<opengl::webgl::PlatformDevice>();
+  IGL_ASSERT(platformDevice != nullptr);
+  drawable = platformDevice->createTextureFromNativeDrawable(&ret);
+
+  IGL_ASSERT_MSG(ret.isOk(), ret.message.c_str());
+  IGL_ASSERT(drawable != nullptr);
+  return drawable;
+}
+
+static void createFramebuffer(const std::shared_ptr<ITexture>& nativeDrawable) {
+  FramebufferDesc framebufferDesc;
+  framebufferDesc.colorAttachments[0].texture = nativeDrawable;
+
+  const TextureDesc desc = TextureDesc::new2D(
+      nativeDrawable->getFormat(),
+      nativeDrawable->getDimensions().width,
+      nativeDrawable->getDimensions().height,
+      TextureDesc::TextureUsageBits::Attachment | TextureDesc::TextureUsageBits::Sampled,
+      IGL_FORMAT("{}C{}", framebufferDesc.debugName.c_str(), 0).c_str());
```

Comment:
It can be `framebufferDesc.debugName.c_str()` now and let's get rid of `IGL_FORMAT` in this file.